### ISSUE TITLE
Knawat Support WPML

### ISSUE THREAD

**Issue by HazemKhaled** [Knawat/dropshipping-woocommerce#154]
We need to create new plugin to support WPML

Acceptance Criteria:

- [ ] Split qTranslate-x code in a separate repo and separate plugin
- [ ] Create new plugin in a separate repo to support WPML
- [ ] Check if qTranslate-x or WPML installed, recommend our new plugins to be installed
- [ ] WPML plugin should support product title, description, attributes, variations attributes, tags, and categories

**Comment by suraj-rathod** [comment]
I have reviewed the WPML plugin and on the initial level, i have considered the plugin support with the addon for knawat dropshipping plugin.

Whoever is using the WPML plugin with knawat dropshipping plugin so will show the notice like there is an addon available.

As of now in the initial version, we can add an option for the products, category, tags, attributes, and variations translation but in the latter part we can discuss other challenges related to translation it which describe below.

1. What if a user will use WPML after importing all products with qTranslate-X
2. We have to give any option related to the switch over the plugin from between two
3. qTranslate-X plugin create only a single product with a language string parameter
4. WPML create a replica of the item based on the primary language of the site

**Comment by iamdharmesh** [comment]
@suraj-rathod okay please start on develop addon.

> 1. What if a user will use WPML after importing all products with qTranslate-X 
> 2. We have to give any option related to the switch over the plugin from between two
> 3. qTranslate-X plugin create only a single product with a language string parameter
> 4. WPML create a replica of the item based on the primary language of the site

for this I think its related to migration between 2 plugin but its rare and also it not looks blocker for now as we can ask customer to delete products and re-import using new translation plugin. 

@HazemKhaled do you have any thoughts on listed questions?

**Comment by HazemKhaled** [comment]
Agree with @iamdharmesh, Migration is not our responsibility, not switching from any plugin to another

**Comment by iamdharmesh** [comment]
@HeshamIsmail addon for WPML is already developed please download it from here https://github.com/Knawat/dropshipping-woocommerce-wpml-addon test please.

**Comment by HeshamIsmail** [comment]
@iamdharmesh sure, on it

**Comment by HeshamIsmail** [comment]
Our test met the acceptance criteria set above 

![image](https://user-images.githubusercontent.com/74185849/112001870-668ade00-8b30-11eb-804e-5cedd82064f4.png)
![image](https://user-images.githubusercontent.com/74185849/112001916-6f7baf80-8b30-11eb-82b6-fae1693f179a.png)
![image](https://user-images.githubusercontent.com/74185849/112001941-75719080-8b30-11eb-8658-e32af81c5c9c.png)

**Comment by HazemKhaled** [comment]
Where is the repo for Knawat qTranslate-X addon?

Tests:

1. Install WPML, then Install Knawat >>> should ask you to install Knawat WPML Addon
2. Install qTranslate-X then install Knawat >>> should ask you to install Knawat qTranslate-x Addon
3. Install Knawat only >>> Should not ask you to install any of them
4. Install old Knawat plugin with qTranslate-x, then upgrade to the dev version >>> should ask you to install Knawat qTranslate-x Addon

You need also to test import, and re-import products, make sure all texts are coming as before, and translated if a translation configured

**Comment by iamdharmesh** [comment]
@HazemKhaled As discussed over Slack, we didn't created separate addon for qTranslate-x, its part of plugin. while You can find WPML addon here https://github.com/Knawat/dropshipping-woocommerce-wpml-addon

**Comment by HeshamIsmail** [comment]
1.Install WPML, then Install Knawat ----> should ask you to install Knawat WPML Addon

Tested this scenario using **WooCommerce Multilingual – run WooCommerce with WPML** instead of **WPML** (Paid)

**Result**: No message requiring the installation of "Knawat WPML Addon"
![image](https://user-images.githubusercontent.com/74185849/113882985-70fed600-97c6-11eb-9cca-8f61d1daf05d.png)

2.Install qTranslate-X then install Knawat-----> should ask you to install Knawat WPML Addon

**Result**: No message requiring the installation of "Knawat qTranslate-x Addon"

![image](https://user-images.githubusercontent.com/74185849/113884138-6abd2980-97c7-11eb-8c5a-3905b1bfc527.png)

3.Install Knawat only -------> Should not ask you to install any of them

**Result**: No message requiring the installation of "Knawat qTranslate-x Addon" or "Knawat WPML Addon"

![image](https://user-images.githubusercontent.com/74185849/113885196-53327080-97c8-11eb-885c-63b18d703e74.png)

4.Install old Knawat plugin with qTranslate-x, then upgrade to the dev version------> should ask you to install Knawat qTranslate-x Addon

**Result**: WordPress is requiring installing the ordinary qTranslate-x

![image](https://user-images.githubusercontent.com/74185849/113886467-5d08a380-97c9-11eb-8060-f6cef88c33ce.png)

**Comment by HeshamIsmail** [comment]
As per Dharmesh comment the acceptance criteria is met 

![image](https://user-images.githubusercontent.com/74185849/113887063-d7d1be80-97c9-11eb-920f-8f27eac8558b.png)

@HazemKhaled please approve

**Comment by HazemKhaled** [comment]
> @HazemKhaled As discussed over Slack, we didn't created separate addon for qTranslate-x, its part of plugin. while You can find WPML addon here https://github.com/Knawat/dropshipping-woocommerce-wpml-addon

No please, qTranslate-x is super outdated, we should decouple it from Knawat plugin pls

**Comment by suraj-rathod** [comment]
> > @HazemKhaled As discussed over Slack, we didn't created separate addon for qTranslate-x, its part of plugin. while You can find WPML addon here https://github.com/Knawat/dropshipping-woocommerce-wpml-addon
> 
> No please, qTranslate-x is super outdated, we should decouple it from Knawat plugin pls

@iamdharmesh @HazemKhaled So we have to remove the qTranslate-x plugin with knawat dropshipping installation and create a separate repository for qTranslate-X addon

**Comment by iamdharmesh** [comment]
Yes @suraj-rathod need to de-couple qTranslate-X and qTranslate-XT related functionality from plugin and create separate add-on for it.

@HazemKhaled do we need to remove qTranslate-X from plugin suggestions in plugin on boarding?

**Comment by iamdharmesh** [comment]
Hello @suraj-rathod @HeshamIsmail ,

Based on discussion with @HazemKhaled we decided to keep qTranslate-X with main plugin and will not make it separate with addon. 

So, we will go with WPML addon only and new acceptance Criteria is like below.
**Acceptance Criteria:**

- [ ] Create new plugin in a separate repo to support WPML
- [ ] Check if WPML installed, recommend our new plugin to be installed
- [ ] WPML plugin should support product title, description, attributes, variations attributes, tags, and categories

Thanks

**Comment by HeshamIsmail** [comment]
My test didn't meet the acceptance criteria below:

 Check if WPML installed, recommend our new plugin to be installed (@suraj-rathod  has created a PR for it waiting to be merged)
 WPML plugin should support product title, description, attributes, variations attributes, tags, and categories (@suraj-rathod  has created a PR for it waiting to be merged)

**Comment by HeshamIsmail** [comment]
The product count is detecting the different supported language as required however the titles are disappearing.

@suraj-rathod  is already working on this issue on [](https://github.com/Knawat/dropshipping-woocommerce-wpml-addon/issues/4)

**Comment by suraj-rathod** [comment]
Hello @HeshamIsmail Can you please review it and let us know your feedback.

1) https://www.screencast.com/t/uCZP0HTMgy
2) https://www.screencast.com/t/wxzLwjkW
3) https://www.screencast.com/t/BCLJMFCrZO

**Comment by suraj-rathod** [comment]
@iamdharmesh Can you please review it below screen recorded video as i am facing this issue while updating a record with translation in WMPL plugin screen or  
https://www.screencast.com/t/u61v1tDVQFm8

**Comment by HeshamIsmail** [comment]
As per my last meeting with Suraj, we are still facing issues between WPML and our Add on few of them will only be fixed from WPML which shouldn't delay our release, however, one of the issues regarding titles could be handled from our end, Suraj will work on it and update us on Thursday.

**Comment by suraj-rathod** [comment]
@HeshamIsmail as per the last meeting we discussed over WPML and title issue which is caused by the WPML addon so i have checked plugin files and found one issue where WPML update the title and after that our title was empty so in searching of that i have disabled the code so that things are working fine in multiple runs so right now i am finding the way to remove that defected WPML code or run our function again so that issue not happening as we can not touch or comment any WPML code so i am on it and i will update you over Monday on it.
https://www.screencast.com/t/RyZQPFgz2V1
